File: mifosng-android/src/main/java/com/mifos/mifosxdroid/formwidgets/FormEditText.java
/*
 * This project is licensed under the open source MPL V2.
 * See https://github.com/openMF/android-client/blob/master/LICENSE.md
 */

package com.mifos.mifosxdroid.formwidgets;

import android.content.Context;
import android.support.v4.app.FragmentManager;
import android.view.MotionEvent;
import android.view.View;
import android.view.inputmethod.EditorInfo;
import android.widget.EditText;
import android.widget.TextView;

import com.mifos.mifosxdroid.uihelpers.MFDatePicker;

/**
 * Created by ishankhanna on 01/08/14.
 */
public class FormEditText extends FormWidget {

    protected TextView label;
    protected EditText input;

    private Boolean isDateAvailable;

    public FormEditText(Context context, String name) {

        super(context, name);
        label = new TextView(context);
        label.setText(getDisplayText());
        label.setLayoutParams(FormWidget.defaultLayoutParams);

        input = new EditText(context);
        input.setLayoutParams(FormWidget.defaultLayoutParams);
        input.setImeOptions(EditorInfo.IME_ACTION_DONE);
        isDateAvailable = false;
        layout.addView(label);
        layout.addView(input);


    }

    @Override
    public String getValue() {
        return input.getText().toString();
    }

    @Override
    public void setValue(String value) {
        input.setText(value);
    }

    @Override
    public void setHint(String value) {
        input.setHint(value);
    }

    public Boolean isDateField() {
        return isDateAvailable;
    }

    public void setIsDateField(Boolean isDateField, final FragmentManager fragmentManager) {

        this.isDateAvailable = isDateField;

        if (isDateField()) {

            input.setOnTouchListener(new View.OnTouchListener() {
                @Override
                public boolean onTouch(View v, MotionEvent event) {

                    if (MotionEvent.ACTION_UP == event.getAction()) {

                        final MFDatePicker mfDatePicker = new MFDatePicker();
                        mfDatePicker.setOnDatePickListener(new MFDatePicker.OnDatePickListener() {
                            @Override
                            public void onDatePicked(String date) {
                                setValue(date);
                                mfDatePicker.dismiss();
                            }
                        });
                        mfDatePicker.show(fragmentManager, MFDatePicker.TAG);
                    }

                    return true;
                }
            });


        } else {
            throw new RuntimeException("This EditText must be a Date Field! Please check if " +
                    "you've set isDateAvailable = true or not");
        }

    }


}

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/formwidgets/FormEditText.kt
/*
 * This project is licensed under the open source MPL V2.
 * See https://github.com/openMF/android-client/blob/master/LICENSE.md
 */
package com.mifos.mifosxdroid.formwidgets

import android.content.Context
import android.view.MotionEvent
import android.view.inputmethod.EditorInfo
import android.widget.EditText
import android.widget.TextView
import androidx.fragment.app.FragmentManager
import com.mifos.mifosxdroid.uihelpers.MFDatePicker
import com.mifos.mifosxdroid.uihelpers.MFDatePicker.OnDatePickListener

/**
 * Created by ishankhanna on 01/08/14.
 */
class FormEditText(context: Context?, name: String?) : FormWidget(context, name) {
    private var label: TextView
    private var input: EditText
    var isDateField: Boolean

    init {
        label = TextView(context)
        label.text = displayText
        label.layoutParams = defaultLayoutParams
        input = EditText(context)
        input.layoutParams = defaultLayoutParams
        input.imeOptions = EditorInfo.IME_ACTION_DONE
        isDateField = false
        layout.addView(label)
        layout.addView(input)
    }

    override var value: String
        get() = input.text.toString()
        set(value) {
            input.setText(value)
        }

    override fun setHint(value: String?) {
        input.hint = value
    }

    fun setIsDateField(isDateField: Boolean, fragmentManager: FragmentManager?) {
        this.isDateField = isDateField
        if (this.isDateField) {
            input.setOnTouchListener { v, event ->
                if (MotionEvent.ACTION_UP == event.action) {
                    val mfDatePicker = MFDatePicker()
                    mfDatePicker.onDatePickListener = object : OnDatePickListener {
                        override fun onDatePicked(date: String?) {
                            if (date != null) {
                                value = date
                            }
                            mfDatePicker.dismiss()
                        }
                    }
                    if (fragmentManager != null) {
                        mfDatePicker.show(fragmentManager, MFDatePicker.TAG)
                    }
                }
                true
            }
        } else {
            throw RuntimeException(
                "This EditText must be a Date Field! Please check if " +
                        "you've set isDateAvailable = true or not"
            )
        }
    }
}